app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :likes, class_name: 'Like', foreign_key: 'post_id'
+  has_many :comments, class_name: 'Comment', foreign_key: 'post_id'
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  def update_post_count
+    author.increment!(:posts_counter)
+  end
```

Comment:
- [ ] [OPTIONAL] Good job implementing the "update_counter" method. However, we can make this better by using one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) methods `after_save` to call the method every time a new `post` has been saved to the database, then make this method private.

- [HINT] You can use this method the following way:
```ruby
  after_save :update_post_count

  private

  def update_post_count
    author.increment!(:posts_counter)
  end
```